lib/instrumentation/span.js

```diff
@@ -35,6 +35,7 @@ function Span (transaction, name, ...args) {
   this._destination = null
   this._message = null
   this._stackObj = null
+  this._capturedStackTrace = null
```

Comment:
Below at:

```js
  if (this._agent._conf.captureSpanStackTraces) {
          this._recordStackTrace()
```

should we consider also skipping if `_conf.spanFramesMinDuration <= 0`?  This would allow the same behaviour as the Java agent's https://www.elastic.co/guide/en/apm/agent/java/current/config-stacktrace.html#config-span-frames-min-duration where a negative value (or `0` as you are using in tests below) will turn off the `Error.captureStackTrace()` calls and save some more cycles.  This might be relevant if/when `spanFramesMinDuration` is added to central config.

Reply:
Good question! I did some thinking and testing on this one today.

First -- I think we'll be ok going either way on this.  The behavior [isn't speced](https://github.com/elastic/apm/blob/master/specs/agents/tracing-spans.md#span-stack-traces) so we have some flexibility.

Second -- if [I'm reading those docs correctly](https://www.elastic.co/guide/en/apm/agent/java/current/config-stacktrace.html#config-span-frames-min-duration) -- then zero and negative values actually have different meanings.  Zero is appears to disable traces

> To disable stack trace collection for spans completely, set the value to 0ms.

While negative values indicate every span should have a stack trace

> When setting this option to a negative value, like -1ms, stack traces will be collected for all spans.

It sounds like [this has come up before](https://github.com/elastic/apm/issues/94) and that both Andrew and Felix find the current state of affairs in the java agent confusing.  I tend to agree with them -- a `0` to me means collect stack traces for all the spans.

So, for now my preference would be to skate to where the java agent is headed and just honor `0` as the "always collect stack traces" value since (almost?) all spans should have some positive duration, and allow the existing `captureSpanStackTraces` to be the global toggle instead of fiddling around with a negative values.

Reply:
I agree with "skate to where the java agent is headed" in general.
Are we going move to:

1. Having a negative value meaningful -- to mean "never collect span stack traces" -- and then eventually deprecate `captureSpanStackTraces` in favour of just the single numeric config var; or
2. Continue with two separate vars: (a) a boolean to turn it on or off, plus (b) a numeric to just specify the duration if the boolean is "on"; and suggest other agents do the same.

Reply:
Capturing a decision made elsewhere

1. Implement the behavior as the other agents have (-1 means collect all, and a 0 means collect none)
2. Leave our captureSpanStackTraces boolean as is for backward compatibility reasons.  If captureSpanStackTraces=false then nothing is collected.  If it's true then the value in spanFramesMinDuration is also consulted.
3. Restart the discussion in https://github.com/elastic/apm-agent-nodejs/pull/2081 in order to decide what everyone should do moving forward.